crates/subspace-farmer/src/farm.rs

```diff
@@ -499,6 +501,19 @@ impl FarmId {
     pub fn new() -> Self {
         Self::Ulid(Ulid::new())
     }
+
+    /// Derive sub IDs
+    #[inline]
+    pub fn derive_sub_ids(&self, n: usize) -> Vec<Self> {
```

Comment:
This makes no sense to me, can you explain what this is supposed to mean?

Reply:
Using a child id derived from a farmer or cache avoids the need to transfer farm ids, which would make it redundant to randomize new ids for each farm when the farmer or cache is already randomly generating ids.

Reply:
Using a child id derived from a farmer or cache avoids the need to transfer farm ids, which would make it redundant to randomize new ids for each farm when the farmer or cache is already randomly generating ids.

Reply:
Still makes no sense to me, sorry.

This is farm ID, just one of the many farms the farmer is managing. Other farms have different IDs. I don't understand what this derivation has to do with any of that.

Reply:
I can remove him if you don't think it's necessary, it doesn't affect the overall logic here

Reply:
I'm trying to understand why it was added in the first place and so far it doesn't really make sense to me

Reply:
Oh, I remember, using farmer_id to derive sub_ids here would simply infer that the status of those farms should be updated when the farmer identifies them without any other correlation.

Reply:
Otherwise we need to record these correlations in the controller, or send the included farms_id in the farmer identify

Reply:
That is not `farmer_id` though, the type is `FarmId`, meaning ID of a single farm. It has nothing to do with other farms and doesn't identify them in any way. You can't infer anything about any other farm from it and you certainly need to track that in the controller (in fact it is already done).

Reply:
This does make it confusing, maybe I should change the type to FarmerId.

Reply:
It makes no sense to me either way. You need to transfer the same information that was transferred before these changes anyway.

Reply:
Either way we need to know what farms are inside the farmer, recorded or derived, so you prefer to record the farms_id inside the farmer in the controller.

Reply:
There are already `farm_id`s of individual farms in the controller. What needs to be is to group them by farmer ID somehow and only transfer details once during initialization.

Reply:
I know what you mean. I'll try it later.